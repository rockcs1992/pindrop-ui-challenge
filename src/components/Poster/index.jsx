import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  from {
    transform: translateY(20%);
    opacity: 0;
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

const Image = styled.img.attrs({
  src: ({ imgSrc }) => imgSrc
})`
  height: 100%;
  width: 100%;
  display: block;
`;

const Title = styled.p`
  background-color: lightgrey;
  margin: 0;
  width: 100%;
  text-align: center;
`;

const PosterWrapper = styled.div`
  position: relative;
  display: inline-block;
  animation: ${fadeIn} 0.5s;

  &:hover {
    transform: scale(1.2);
    transition: 0.5s;
    z-index: 99;
  }

  &:hover:after {
    background-color: transparent;
    transition: 0.3s;
  }
  
  &:not(:hover) {
    transition: 0.5s;
  }

  &::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: rgba(0,0,0,0.3);
    left: 0;
    top: 0;
  }
`;

const Poster = ({ imgSrc, title }) => {
  const [imageLoaded, showPoster] = useState(false);
  return (
    <PosterWrapper>
      <Image imgSrc={imgSrc} onLoad={() => {showPoster(true)}}/>
      {imageLoaded && <Title>{title}</Title>}
    </PosterWrapper>
  );
}

Poster.propTypes = {
  imgSrc: PropTypes.string,
  title: PropTypes.string
};

export default Poster;
