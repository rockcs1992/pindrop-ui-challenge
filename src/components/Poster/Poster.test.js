import React from 'react';
import { render } from 'enzyme';
import Poster from './index';

describe('Poster', () => {
    it('should render correctly', () => {
        const tree = render(<Poster />);
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly with img src and title', () => {
        const tree = render(<Poster imgSrc={'test.png'} title={'title'} />);
        expect(tree).toMatchSnapshot();
    });
})