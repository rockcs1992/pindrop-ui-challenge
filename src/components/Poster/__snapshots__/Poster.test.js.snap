// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster should render correctly 1`] = `
<div
  class="sc-htpNat gGkIdj"
>
  <img
    class="sc-bdVaJa kvPpCZ"
  />
</div>
`;

exports[`Poster should render correctly with img src and title 1`] = `
<div
  class="sc-htpNat gGkIdj"
>
  <img
    class="sc-bdVaJa kvPpCZ"
    src="test.png"
  />
</div>
`;
